ide/nativeimage.api/src/org/netbeans/modules/nativeimage/api/debug/NIDebugger.java

```diff
@@ -107,12 +108,55 @@ public void removeBreakpoint(Object id) {
      * @param executionDescriptor execution descriptor that describes the runtime attributes
      * @param startedEngine the corresponding DebuggerEngine is passed to this consumer
      * @return future that completes on the execution finish
-     * @since 1.0
+     * @since 0.1
+     * @deprecated Use {@link #start(org.netbeans.modules.nativeimage.api.debug.StartDebugParameters, java.util.function.Consumer)}
      */
+    @Deprecated
     public CompletableFuture<Void> start(List<String> command, File workingDirectory, String debugger, String displayName, ExecutionDescriptor executionDescriptor, Consumer<DebuggerEngine> startedEngine) {
         return provider.start(command, workingDirectory, debugger, displayName, executionDescriptor, startedEngine);
     }
 
+    /**
+     * Start the actual debugging session. Call this typically after breakpoints are added.
+     *
+     * @param debugParameters parameters to start the debugging with
+     * @param startedEngine the corresponding DebuggerEngine is passed to this consumer
+     * @return a future which is completed when the started debugger session finishes
+     * @since 0.5
+     */
+    public CompletableFuture<Void> start(StartDebugParameters debugParameters, Consumer<DebuggerEngine> startedEngine) {
+        Objects.requireNonNull(debugParameters);
+        CompletableFuture<Void> start = provider.start(debugParameters, startedEngine);
+        if (start.isCompletedExceptionally()) {
+            try {
+                start.get();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It looks like you are using CompletableFuture.get() without setting its timeout parameter. Be sure to set timeout values or handle timeout exceptions to prevent threads from waiting for a process to end. This can lead to downtime for your customers. [here](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/CompletableFuture.html#get-long-java.util.concurrent.TimeUnit-).